### DISCUSSION TITLE
About Static HTML Export

### DISCUSSION THREAD

**Discussion by hanlray** [vercel/next.js#13806] [Help] [answered]
I'm using the "Static HTML Export" feature to generate a static html site. I deployed those files to CDN and not running any server. I have a link to the 'about' page in the home page, so I wrote in index.js

`<Link href="/about"><a>about</a></Link>`

This is fine when I clicked the about link on the homepage. But when I tried to load the about page directly on the browser address bar such as "http://www.example.com/about", I got 404 error. I realized that the reason is that next/link component is client-side navigation. So I changed the link to:

`<Link href="/about" as="/about.html"><a>about</a></Link>`

But this time, the link does not work in the developing mode via "yarn dev".

Any suggestion? Thanks.

**Comment by jamesmosier** [comment, marked as answer]
Hi @hanlray. Do you have code you could share? Or additionally, what CDN/static host are you using?

**Reply by hanlray** [reply]
Sorry, I have no code available currently. I'm using the CDN from tencent in China.

**Reply by Moto-Venturer** [reply]
@jamesmosier if you are looking still looking for a solid CDN/static host, check out 21YunBox.  They also had a solid english support team.  Good luck!

**Comment by Heein-Park** [comment]
In my similar case, the exported HTML files were generated as "about.html" in the directory "about." But in the usual web server, the page that a user accesses via the browser address bar indicates "index.html" in the directory "about." The links in webpages refer to the about.html directly, so they seem to behave normally, but it will cause problems in your and my cases.

https://nextjs.org/docs/api-reference/next.config.js/exportPathMap#adding-a-trailing-slash

I resolved this issue by adding
module.exports = {
  exportTrailingSlash: true,
}
into next.confing.js.

It will generate about/index.html instead of about.html or about/about.html (this statement will not be accurate.)

I believed that there will be more effective solution, such as changing the behavior of your web server.

**Reply by hanlray** [reply]
Sound reasonable, I'll try it. Thank you.

**Reply by banshee** [reply]
This option has been renamed to trailingSlash, FYI

**Reply by OttoG** [reply]
Thank you. I noticed that for this to work properly, links need to be written as absolute paths. (I.e., with a leading slash.)
